Spigot-Server-Patches/0381-Call-player-spectator-target-events.patch

```diff
@@ -0,0 +1,51 @@
+From 56358236f603519affbb0d600cd4efc6147763af Mon Sep 17 00:00:00 2001
+From: Caleb Bassham <caleb.bassham@gmail.com>
+Date: Fri, 28 Sep 2018 02:32:19 -0500
+Subject: [PATCH] Call player spectator target events
+
+
+diff --git a/src/main/java/net/minecraft/server/EntityPlayer.java b/src/main/java/net/minecraft/server/EntityPlayer.java
+index 5ab4e01ed..3283ef215 100644
+--- a/src/main/java/net/minecraft/server/EntityPlayer.java
++++ b/src/main/java/net/minecraft/server/EntityPlayer.java
+@@ -1373,15 +1373,34 @@ public class EntityPlayer extends EntityHuman implements ICrafting {
+         return (Entity) (this.cv == null ? this : this.cv);
+     }
+ 
+-    public void setSpectatorTarget(Entity entity) {
++    public void setSpectatorTarget(Entity newSpectatorTarget) {
+         Entity entity1 = this.getSpecatorTarget();
+ 
+-        this.cv = (Entity) (entity == null ? this : entity);
+-        if (entity1 != this.cv) {
++        if (newSpectatorTarget == null) {
++            this.cv = this;
```

Comment:
all of these assignments to `this.cv` should probably use an obfhelper setter, with that setter up by the field declaration.
```java
private Entity cv; private void setSpectatorTarget(Entity e) { this.cv = e; } // Paper - OBFHELPER
```

Just so when we go to update this to 1.14 or 1.5512.4 or whatever its clear what the intent was even when this gets renamed.

Reply:
`setSpectatorTarget(Entity)` is the method that I am changing for this pr. Should I do something like `setSpectatorTargetField(Entity)` as an obfhelper or keep it as it is?

Reply:
Naming it slightly different is probably smart, I think that'd be fine 👍